modules/smartxBidAdapter.js

```diff
@@ -0,0 +1,401 @@
+        for (var i = 0; i < bid.params.user.data.length; i++) {
+          var isemq = (bid.params.user.data[i].name) || 'empty';
+          if (isemq !== 'empty') {
+            var provider = bid.params.user.data[i].name;
+            var targetingstring = (bid.params.user.data[i].segment[0].value) || 'empty';
+            targetingarr.push({
+              id: provider,
+              name: provider,
+              segment: {
+                name: provider,
+                value: targetingstring,
+              }
+            })
+          }
+        }
+
+        requestPayload.user = {
+          ext: userExt,
+          data: targetingarr
+        }
+      }
+
+      return {
+        method: 'POST',
+        url: URL,
+        data: requestPayload,
+        bidRequest: bidderRequest,
+        options: {
+          contentType: 'application/json',
+          customHeaders: {
+            'x-openrtb-version': '2.3'
+          }
+        }
+      };
+    });
+
+    return smartxRequests;
+  },
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, bidderRequest) {
+    const bidResponses = [];
+    const serverResponseBody = serverResponse.body;
+    if (serverResponseBody && utils.isArray(serverResponseBody.seatbid)) {
+      utils._each(serverResponseBody.seatbid, function (bids) {
+        utils._each(bids.bid, function (smartxBid) {
+          let currentBidRequest = {};
+          for (let i in bidderRequest.bidRequest.bids) {
+            if (smartxBid.impid == bidderRequest.bidRequest.bids[i].bidId) {
+              currentBidRequest = bidderRequest.bidRequest.bids[i];
+            }
+          }
+          /**
+           * Make sure currency and price are the right ones
+           * TODO: what about the pre_market_bid partners sizes?
+           */
+          utils._each(currentBidRequest.params.pre_market_bids, function (pmb) {
+            if (pmb.deal_id == smartxBid.id) {
+              smartxBid.price = pmb.price;
+              serverResponseBody.cur = pmb.currency;
+            }
+          });
+          const bid = {
+            requestId: currentBidRequest.bidId,
+            currency: serverResponseBody.cur || 'USD',
+            cpm: smartxBid.price,
+            creativeId: smartxBid.crid || '',
+            ttl: 360,
+            netRevenue: true,
+            vastContent: smartxBid.adm,
+            vastXml: smartxBid.adm,
+            mediaType: VIDEO,
+            width: smartxBid.w,
+            height: smartxBid.h
+          };
+          const context1 = utils.deepAccess(currentBidRequest, 'mediaTypes.video.context');
+          const context2 = utils.deepAccess(currentBidRequest, 'params.ad_unit');
+          if (context1 == 'outstream' || context2 == 'outstream') {
```

Comment:
I overlooked those. Triple equals again is safer.

Why have two options configuring the `context` ? What happens when the `context2` is `oustream` but the others are `instream` ?  Then a player is loaded, but everything is configured for `instream` ?

Maybe this parameter can be removed all together?

Reply:
Removed context2, as we are not using params.ad_unit for context identifier.